src/workerd/io/worker-interface.capnp

```diff
@@ -138,6 +177,275 @@ struct Trace @0x8e8d911203762d34 {
 
   truncated @24 :Bool;
   # Indicates that the trace was truncated due to reaching the maximum size limit.
+
+  # The following structs are used only in Streaming Traces.
+
+  # The Onset struct is always sent as the first event in any trace stream. It
+  # contains general metadata about the pipeline that is being traced.
+  struct Onset {
```

Comment:
Nit: I don't think onset is descriptive enough, I would prefer something like TraceMetadata or PipelineTraceMetadata but I know names are personal so I'll leave this as a nit and defer to you.

Reply:
For now I'll keep this an `Onset`. We can revisit specific naming as we move forward tho. It'll be simple to change if necessary before this goes live.